### PR TITLE
WIP drivers: espi: Propagate eSPI host DnX warning to driver's client

### PR DESCRIPTION
**Pull request by albertofloyd**
Propagate Download and Execute (DnX) entry warning.
Add missing handler for SUS warning power down ack.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

**Review by scottwcpg — APPROVED** [review]

**Comment by albertofloyd** [comment]
@scottwcpg seems MEC_ESPI_MSVW08_SRC1_VAL is missing in HAL.

Also do you have any thoughts/preferences in how to update eSPi driver initialization to allow to enable other GIRQ related to virtual wires i.e. GIRQ27

Currently, inside espi_xec_init enables key VW taken from device tree (GIRQ24) and stored in driver configuration structures.

	/* Enable aggregated block interrupts for VWires */
	MCHP_GIRQ_ENSET(config->vw_girq_id) = MEC_ESPI_MSVW00_SRC0_VAL |
		MEC_ESPI_MSVW00_SRC1_VAL | MEC_ESPI_MSVW00_SRC2_VAL |
		MEC_ESPI_MSVW01_SRC1_VAL | MEC_ESPI_MSVW01_SRC2_VAL |
		MEC_ESPI_MSVW02_SRC0_VAL | MEC_ESPI_MSVW03_SRC0_VAL;

**Comment by scottwcpg** [comment]
The HAL is blocked by Google. This is a huge issue.
I would hard code the values you need in the driver for now.
You could also hard code the GIRQ27 stuff in the driver and let us clean it up when the HAL and MEC172x stuff can be pushed.
MEC172x has the same VWires as MEC152x.

**Comment by albertofloyd** [comment]
> x has the same VWires as MEC152x.

I thought only MEC17x is the problem, are you saying rather fixing in both places once discussion concludes?
If that's the case, I will uUse macros for missing MSV008 definitions in driver for now,  replace when HAL makes it available.

For blocks IRQ extension, I was thinking to use device tree, so no HAL dependency there.

**Comment by scottwcpg** [comment]
@albertofloyd The HAL repo contains MEC152x now.  If I make a MEC152x specific change I hope Google will not see it and interfere.  If you want the MEC152x header update I will do it and hope it goes through.
I agree with our for Device Tree for the GIRQ's and any other items we can avoid HAL header changes.

File: drivers/espi/espi_mchp_xec.c [hunk 1]
```diff
@@ -21,6 +21,10 @@
  */
 #define ESPI_XEC_VWIRE_SEND_TIMEOUT 100ul
 
+#define VW_MAX_GIRQS                2ul
+/* Missing HAL value */
+#define MEC_ESPI_MSVW08_SRC1_VAL    5ul
```
Comment (albertofloyd) [inline drivers/espi/espi_mchp_xec.c:26]: @scottwcpg this is the missing HAL value. I have updated the PR with device tree changes

**Review by nashif — APPROVED** [review]

**Comment by albertofloyd** [comment]
@nashif could you help revert this or advise how to proceed? This was WIP and it has been merged to "main" not master

**Comment by albertofloyd** [comment]
> @nashif could you help revert this or advise how to proceed? This was WIP and it has been merged to "main" not master

@nashif Ignore this just learned about branch renaming

**Pull request merged**